fastlane/lib/fastlane/environment_printer.rb

```diff
@@ -0,0 +1,192 @@
+        unless Fastlane::TOOLS.include?(x.name.to_sym)
+          table << "| #{x.name} | #{x.version} |\n"
+        end
+      end
+      rendered_table = MarkdownTableFormatter.new table
+
+      env_output << rendered_table.to_md
+      env_output << "</details>\n\n"
+      return env_output
+    end
+
+    def self.print_system_environment
+      require "openssl"
+
+      env_output = "### Stack\n\n"
+      product, version, build = `sw_vers`.strip.split("\n").map { |line| line.split(':').last.strip }
+      table_content = {
+        "fastlane" => Fastlane::VERSION,
+        "OS" => `sw_vers -productVersion`.strip,
+        "Ruby" => RUBY_VERSION,
+        "Bundler?" => Helper.bundler?,
+        "Xcode Path" => Helper.xcode_path,
+        "Xcode Version" => Helper.xcode_version,
+        "Git" => `git --version`.strip.split("\n").first,
+        "Installation Source" => $PROGRAM_NAME,
+        "Host" => "#{product} #{version} (#{build})",
+        "Ruby Lib Dir" => RbConfig::CONFIG['libdir'],
+        "OpenSSL Version" => OpenSSL::OPENSSL_VERSION
+      }
+      table = ["| Key | Value |"]
+      table += table_content.collect { |k, v| "| #{k} | #{v} |" }
+
+      begin
+        rendered_table = MarkdownTableFormatter.new(table.join("\n"))
+        env_output << rendered_table.to_md
+      rescue => ex
+        UI.error(ex)
+        UI.error("Error rendering markdown table using the following text:")
+        UI.message(table.join("\n"))
+        env_output << table.join("\n")
+      end
+
+      env_output << "\n\n"
+      env_output
+    end
+
+    def self.print_fastlane_files
+      env_output = "### fastlane files:\n\n"
+
+      fastlane_path = FastlaneFolder.fastfile_path
+
+      if fastlane_path && File.exist?(fastlane_path)
+        env_output << "<details>"
+        env_output << "<summary>`#{fastlane_path}`</summary>\n"
+        env_output << "\n"
+        env_output << "```ruby\n"
+        env_output <<  File.read(fastlane_path)
+        env_output <<  "```\n"
+        env_output << "</details>"
+      else
+        env_output << "**No Fastfile found**\n"
+      end
+      env_output << "\n\n"
+
+      appfile_path = CredentialsManager::AppfileConfig.default_path
+      if appfile_path && File.exist?(appfile_path)
+        env_output << "<details>"
+        env_output << "<summary>`#{appfile_path}`</summary>\n"
+        env_output << "\n"
+        env_output << "```ruby\n"
+        env_output <<  File.read(appfile_path)
+        env_output <<  "```\n"
+      else
+        env_output <<  "**No Appfile found**\n"
+      end
+      env_output << "\n\n"
+      env_output
+    end
+
+    # Copy a given string into the clipboard
+    # Make sure to ask the user first, as some people don't
+    # use a clipboard manager, so they might lose something important
+    def self.copy_to_clipboard(string)
+      require 'tmpdir'
+      tmp_file = File.join(Dir.mktmpdir, "temporary") # so that we don't have problems with escaping the string
```

Comment:
Thanks, I updated the code. Note that `tmp_file` is not the path, but a `File` object, I had to change the code to use `tmp_file.path` at least on the shelling out part